packages/kerosene-ui/src/hooks/useRect.ts

```diff
@@ -25,9 +25,11 @@ const DEFAULT_RECT: Rect = {
 /**
  * Custom React Hook for reading bounding rect of a DOM Element
  * @param disable When set to `true`, updating the rect will be disabled
+ * @param event Listens for specified event type and triggers update, default is "scroll"
  */
 export default function useRect(
   disable = false,
+  event = "scroll",
```

Comment:
`scroll` is a very noisy event, are there any performance implications here?

It doesn't look obvious to me why we want `scroll` as the default

Reply:
`scroll` can be noisy, but along with `resize` it is one of the more common events that can change the bounding rect of an element. We're also returning the window scroll position as part of this hook as I've found that it is very commonly used together with the rect of an element.
The `update` function here is throttled with `requestAnimationFrame`, so will be called at most once per frame.
The reason for this PR was to add the capability to optionally listen for `transitionend` events, but to make this easier to use with any additional events.